app/models/user.rb

```diff
@@ -38,4 +38,16 @@ def user_total_hours_mentored
   def unique_mentees
     pairees.uniq
   end
+
+  def self.active_pebbles(id)
+  where('users.id IN (SELECT rock_and_pebbles.pebble_id FROM users join rock_and_pebbles
+         on rock_and_pebbles.rock_id = users.id WHERE rock_and_pebbles.active = true AND 
+         rock_and_pebbles.rock_id = ?)', id)
+  end 
+
+  def self.active_rocks(id)
+    where('users.id IN (SELECT rock_and_pebbles.rock_id FROM users join rock_and_pebbles
+    on rock_and_pebbles.rock_id = users.id WHERE rock_and_pebbles.active = true AND 
+    rock_and_pebbles.pebble_id = ?)', id)
+  end 
```

Comment:
Nice sub-querying!

I wonder if, because you have to pass in the ID for a specific user anyway, these would look slightly cleaner as instance methods? (I think `id` would be available by virtue of being another instance method at the same scope, but if not `self.id` would grab it)

```suggestion
  def active_pebbles
  where('users.id IN (SELECT rock_and_pebbles.pebble_id FROM users join rock_and_pebbles
         on rock_and_pebbles.rock_id = users.id WHERE rock_and_pebbles.active = true AND 
         rock_and_pebbles.rock_id = ?)', id)
  end 

  def active_rocks
    where('users.id IN (SELECT rock_and_pebbles.rock_id FROM users join rock_and_pebbles
    on rock_and_pebbles.rock_id = users.id WHERE rock_and_pebbles.active = true AND 
    rock_and_pebbles.pebble_id = ?)', id)
  end 
```

Reply:
Made an instance method.